docs/references/phpdoc/tags/property.rst

```diff
@@ -1,34 +1,49 @@
-@property
-=========
+@property, @property-read, @property-write
+===========================================
 
-The @property tag allows a class to know which 'magic' properties are present.
+The ``@property`` tag is used to declare which "magic" `properties`_ are supported.
 
 Syntax
 ------
 
-    @property [Type] [name] [<description>]
+.. code-block::
+
+    @property[<-read|-write>] [Type] [name] [<description>]
 
 Description
 -----------
 
-The @property tag is used in the situation where a class contains the
-``__get()`` and ``__set()`` magic methods and allows for specific names.
+The ``@property`` tag is used when a ``class`` or ``trait`` implements the
+`__get() <https://www.php.net/language.oop5.overloading#object.get>`_ and/or
+`__set() <https://www.php.net/language.oop5.overloading#object.set>`_ "magic"
+methods to resolve non-literal `properties`_ at run-time.
+
+The ``@property-read`` and ``@property-write`` variants MAY be used to indicate "magic"
+properties that can only be read or written.
+
+For example, the ``@property-read`` tag could be used when a class contains
+a `__get() <https://www.php.net/language.oop5.overloading#object.get>` magic
+method which allows for specific names, while those names are not covered in the
+`__set() <https://www.php.net/language.oop5.overloading#object.set>`_ magic method.
 
-An example of this is a child class whose parent has a __get(). The child knows
-which properties need to be present but relies on the parent class to use the
-__get() method to provide it.
-In this situation, the child class would have a @property tag for each magic
-property.
+============================ =====================
+Property supported via       Tag to use
+============================ =====================
+``__get()`` and ``__set()``  ``@property``
+``__get()`` only             ``@property-read``
+``__set()`` only             ``@property-write``
+============================ =====================
 
-@property tags MUST NOT be used in a PHPDoc that is not associated with
-a *class* or *interface*.
+The ``@property``, ``@property-read`` and ``@property-write`` tags can ONLY be used
+in a PHPDoc that is associated with a *class* or *trait*.
```

Comment:
Add `interface` to this list?

Reply:
Eh... interfaces cannot declare properties, so these tags don't apply.

Reply:
Ah, it's _constants_ that I'm thinking of on interfaces.  Ok, then `*interface*` can be removed from line 43 below... or maybe _replaced_ by `*trait*`.

Reply:
Good catch, not sure how I missed that one. Fixed now.